### DISCUSSION TITLE
joining Slack

### DISCUSSION THREAD

**Discussion by ghost** [onnx/onnx#4614] [Q&A] [answered]
Hi,

I need to join either the Arch & Infra SIG or the Converters SIG, and probably both. The problem is that when I login (with my Google account), I get the message that I do not have an account with that workspace.

I would like to join these workspaces because I am working on parsing ONNX in Lean (for theorem proving), and they seem to be the best places to get my questions answered.

Thanks for your attention,

Martin

**Comment by jcwchen** [comment, marked as answer]
Hi @egLanghaus,
Are you able to use this [link](https://join.slack.com/t/lfaifoundation/shared_invite/zt-o65errpw-gMTbwNr7FnNbVXNVFkmyNA) to join Slack? (See [document](https://github.com/onnx/onnx#discuss))

**Comment by ghost** [comment]
Yes, I was able to join, but now that I have your attention:

I want to build a Lean (functional language geared toward theorem proving)
library for parsing ONNX and ONNX-ML files. The ONNX documentation is
sparse and sort of written for someone who already understands it...where
should I start? Google's protobuf documentation? A working example from
another language like Python? If the latter, can you point me to specific
files where I can see both the structure of the ONNX files and some code
that ingests and parses them, building the appropriate objects in the
target language? One challenge seems to be that ONNX files are not totally
human readable, so it has been hard for me to get a handle on it.

Any pointers would be much appreciated!

Martin

**Reply by jcwchen** [reply]
Hi Martin,
ONNX just launched a tutorial website recently: https://onnx.ai/onnx/tutorial_python/index.html. It should be a good starter. In addition, here are the document for [intermediate representation](https://github.com/onnx/onnx/blob/main/docs/IR.md) and [example code](https://github.com/onnx/onnx/blob/main/docs/PythonAPIOverview.md).

**Comment by ghost** [comment]
Killer, thanks!
